# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1385765415 18000
#      Fri Nov 29 17:50:15 2013 -0500
# Branch stable
# Node ID 482cc05b4f99e469d4d1ccc05a39b7602cc16b73
# Parent  8695fa0482267c2008a9fb5b005f9db10ddb9a1c
update gui info text and corresponding NEWS entry

* main-window.cc (main_window::show_gui_info): Updat text.
* NEWS: Use variation of show_gui_info text for entry about the GUI.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -1,16 +1,46 @@
 Summary of important user-visible changes for version 3.8:
 ---------------------------------------------------------
 
- ** An experimental GUI is now available with Octave.  The GUI works well
-    on Linux systems, acceptably on Windows, but unsatisfactorily on Mac OS X.
-    To start a GUI session include the "--force-gui" option when invoking
-    Octave.  The GUI is under continuous improvement and will be the default
-    interface for the 4.0 release.
+  ** One of the biggest new features for Octave 3.8 is a graphical user
+     interface.  It is the one thing that users have requested most
+     often over the last few years and now it is almost ready.  But
+     because it is not quite as polished as we would like, we have
+     decided to wait until the 4.0.x release series before making the
+     GUI the default interface (until then, you can use the --force-gui
+     option to start the GUI).
+
+     Given the length of time and the number of bug fixes and
+     improvements since the last major release Octave, we also decided
+     against delaying the release of all these new improvements any
+     longer just to perfect the GUI.  So please enjoy the 3.8 release of
+     Octave and the preview of the new GUI.  We beleive it is working
+     reasonably well, but we also know that there are some obvious rough
+     spots and many things that could be improved.
+
+     WE NEED YOUR HELP.  There are many ways that you can help us fix
+     the remaining problems, complete the GUI, and improve the overall
+     user experience for both novices and experts alike:
+
+       * If you are a skilled software developer, you can help by
+         contributing your time to help with Octave's development.  See
+         http://octave.org/get-involved.html for more information.
+
+       * If Octave does not work properly, you are encouraged
+         report the problems you find.  See http://octave.org/bugs.html
+         for more information about how to report problems.
+
+       * Whether you are a user or developer, you can help to fund the
+         project.  Octave development takes a lot of time and expertise.
+         Your contributions help to ensure that Octave will continue to
+         improve.  See http://octave.org/donate.html for more details.
+
+    We hope you find Octave to be useful.  Please help us make it even
+    better for the future!
 
  ** Octave now uses OpenGL graphics by default with FLTK widgets.  If
     OpenGL libraries or FLTK widgets are not available when Octave is
     built, gnuplot is used.  You may also choose to use gnuplot for
     graphics by executing the command
 
       graphics_toolkit ("gnuplot")
 
diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -1800,16 +1800,17 @@ main_window::construct_warning_bar (void
   QTextBrowser *msg = new QTextBrowser (box);
   msg->setOpenExternalLinks (true);
   msg->setText
     (tr ("<strong>You are using a release candidate of Octave's experimental GUI.</strong>  "
          "Octave is under continuous improvement and the GUI will be the "
          "default interface for the 4.0 release.  For more information, "
          "select the \"Release Notes\" item in the \"Help\" menu of the GUI, "
          "or visit <a href=\"http://octave.org\">http://octave.org</a>."));
+
   msg->setStyleSheet ("background-color: #ffd97f; color: black; margin 4px;");
   msg->setMinimumWidth (100);
   msg->setMinimumHeight (60);
   msg->setMaximumHeight (80);
   msg->setSizePolicy (QSizePolicy (QSizePolicy::Expanding,
                                    QSizePolicy::Minimum));
 
   QPushButton *info_button = new QPushButton (tr ("More Info"), box);
@@ -1884,24 +1885,51 @@ main_window::hide_warning_bar (void)
 
   construct_gui_info_button ();
 }
 
 void
 main_window::show_gui_info (void)
 {
   QString gui_info
-    (tr ("<p><strong>You are using a release candidate of Octave's experimental GUI.</strong>  "
-         "Octave is under continuous improvement and the GUI will be the "
-         "default interface for the 4.0 release.  For more information, "
-         "select the \"Release Notes\" item in the \"Help\" menu of the GUI, "
-         "or visit <a href=\"http://octave.org\">http://octave.org</a>.</p>"
-         "<p>This message should have more information than the initial "
-         "warning bar message.  Someone needs to write it before we "
-         "create the first release candidate.</p>"));
+    (tr ("<p><strong>A Note about Octave's New GUI</strong></p>"
+         "<p>One of the biggest new features for Octave 3.8 is a graphical "
+         "user interface.  It is the one thing that users have requested "
+         "most often over the last few years and now it is almost ready.  "
+         "But because it is not quite as polished as we would like, we "
+         "have decided to wait until the 4.0.x release series before "
+         "making the GUI the default interface.</p>"
+         "<p>Given the length of time and the number of bug fixes and "
+         "improvements since the last major release Octave, we also "
+         "decided against delaying the release of all these new "
+         "improvements any longer just to perfect the GUI.  So please "
+         "enjoy the 3.8 release of Octave and the preview of the new GUI.  "
+         "We beleive it is working reasonably well, but we also know that "
+         "there are some obvious rough spots and many things that could be "
+         "improved.</p>"
+         "<p><strong>We Need Your Help</strong></p>"
+         "<p>There are many ways that you can help us fix the remaining "
+         "problems, complete the GUI, and improve the overall user "
+         "experience for both novices and experts alike (links will open "
+         "an external browser):</p>"
+         "<p><ul><li>If you are a skilled software developer, you can "
+         "help by contributing your time to help "
+         "<a href=\"http://octave.org/get-involved.html\">develop "
+         "Octave</a>.</li>"
+         "<li>If Octave does not work properly, you are encouraged to "
+         "<a href=\"http://octave.org/bugs.html\">report problems </a> "
+         "that you find.</li>"
+         "<li>Whether you are a user or developer, you can "
+         "<a href=\"http://octave.org/donate.html\">help to fund the "
+         "project</a>.  "
+         "Octave development takes a lot of time and expertise.  "
+         "Your contributions help to ensure that Octave will continue "
+         "to improve.</li></ul></p>"
+         "<p>We hope you find Octave to be useful.  Please help us make "
+         "it even better for the future!</p>"));
 
   QMessageBox gui_info_dialog (QMessageBox::Warning,
                                tr ("Experimental GUI Info"),
                                gui_info, QMessageBox::Close);
 
   gui_info_dialog.exec ();
 }
 
